src/calendar/view/CalendarViewModel.ts

```diff
@@ -60,102 +70,122 @@ export class CalendarViewModel implements EventDragHandlerCallbacks {
 	private previewedEvent: { event: CalendarEvent; model: CalendarEventPreviewViewModel | null } | null = null
 
 	_hiddenCalendars: Set<Id>
-	readonly _calendarInvitations: ReceivedGroupInvitationsModel<GroupType.Calendar>
-	readonly _calendarModel: CalendarModel
-	readonly _entityClient: EntityClient
 	// Events that have been dropped but still need to be rendered as temporary while waiting for entity updates.
 	readonly _transientEvents: Array<CalendarEvent>
-	_draggedEvent: DraggedEvent | null
-	readonly _redrawStream: Stream<void>
+	_draggedEvent: DraggedEvent | null = null
+	private readonly _redrawStream: Stream<void> = stream()
 	readonly _deviceConfig: DeviceConfig
 
 	constructor(
 		private readonly logins: LoginController,
 		private readonly createCalendarEventEditModel: CalendarEventEditModelsFactory,
 		private readonly createCalendarEventPreviewModel: CalendarEventPreviewModelFactory,
-		calendarModel: CalendarModel,
-		entityClient: EntityClient,
+		private readonly calendarModel: CalendarModel,
+		private readonly entityClient: EntityClient,
 		eventController: EventController,
-		progressTracker: ProgressTracker,
-		deviceConfig: DeviceConfig,
-		calendarInvitations: ReceivedGroupInvitationsModel<GroupType.Calendar>,
+		private readonly progressTracker: ProgressTracker,
+		private readonly deviceConfig: DeviceConfig,
+		private readonly calendarInvitationsModel: ReceivedGroupInvitationsModel<GroupType.Calendar>,
 		private readonly timeZone: string,
 	) {
-		this._calendarModel = calendarModel
-		this._entityClient = entityClient
 		this._transientEvents = []
 
 		const userId = logins.getUserController().user._id
 
 		this._deviceConfig = deviceConfig
 		this._hiddenCalendars = new Set(this._deviceConfig.getHiddenCalendars(userId))
-		this.selectedDate = stream(getStartOfDay(new Date()))
-		this._redrawStream = stream()
-		this._draggedEvent = null
-		this._calendarInvitations = calendarInvitations
+
+		this.selectedDate.map((newDate) =>
+			this.preloadMonthsAroundSelectedDate((event) => {
+				const { start, end } = getDayRange(newDate, this.timeZone)
+				return clipRanges(getEventStart(event, this.timeZone).getTime(), getEventEnd(event, this.timeZone).getTime(), start, end) == null
+			}),
+		)
+		this.calendarModel.getEventsForMonths().map((newEvents) =>
+			this.preloadMonthsAroundSelectedDate((event) => {
+				const startOfDay = getStartOfDayWithZone(event.startTime, this.timeZone)
+				return newEvents.get(startOfDay.getTime())?.find((v) => isSameEventInstance(v, event)) == null
```

Comment:
why do we need to detect if it's obsolete in this way? wouldn't we get `DELETE` anyway?